frontend/packages/console-app/src/components/data-view/ResourceDataView.tsx

```diff
@@ -0,0 +1,207 @@
+import { InnerScrollContainer, Tbody, Td, Tr } from '@patternfly/react-table';
+import { useTranslation } from 'react-i18next';
+import {
+  ColumnLayout,
+  K8sResourceCommon,
+} from '@console/dynamic-plugin-sdk/src/extensions/console-types';
+import { createColumnManagementModal } from '@console/internal/components/modals';
+import { TableColumn } from '@console/internal/module/k8s';
+import { EmptyBox } from '@console/shared/src/components/empty-state/EmptyBox';
+import { StatusBox } from '@console/shared/src/components/status/StatusBox';
+import { DataViewLabelFilter } from './DataViewLabelFilter';
+import { ResourceFilters, GetDataViewRows } from './types';
+import { useResourceDataViewData } from './useResourceDataViewData';
+import { useResourceDataViewFilters } from './useResourceDataViewFilters';
+
+export type ResourceDataViewProps<TData, TCustomRowData, TFilters> = {
+  label?: string;
+  data: TData[];
+  loaded: boolean;
+  loadError?: any;
+  columns: TableColumn<TData>[];
+  columnLayout?: ColumnLayout;
+  columnManagementID?: string;
+  initialFilters: TFilters;
+  additionalFilterNodes?: React.ReactNode[];
+  matchesAdditionalFilters?: (resource: TData, filters: TFilters) => boolean;
+  getDataViewRows: GetDataViewRows<TData, TCustomRowData>;
+  customRowData?: TCustomRowData;
+  showNamespaceOverride?: boolean;
+  hideNameLabelFilters?: boolean;
+  hideLabelFilter?: boolean;
+  hideColumnManagement?: boolean;
+  mock?: boolean;
+};
+
+/**
+ * Console DataView component based on PatternFly DataView.
+ */
+export const ResourceDataView = <
+  TData extends K8sResourceCommon = K8sResourceCommon,
+  TCustomRowData = any,
+  TFilters extends ResourceFilters = ResourceFilters
+>({
+  label,
+  data,
+  loaded,
+  loadError,
+  columns,
+  columnLayout,
+  columnManagementID,
+  initialFilters,
+  additionalFilterNodes,
+  matchesAdditionalFilters,
+  getDataViewRows,
+  customRowData,
+  showNamespaceOverride,
+  hideNameLabelFilters,
+  hideLabelFilter,
+  hideColumnManagement,
+  mock,
+}: ResourceDataViewProps<TData, TCustomRowData, TFilters>) => {
+  const { t } = useTranslation();
+
+  const { filters, onSetFilters, clearAllFilters, filteredData } = useResourceDataViewFilters<
+    TData,
+    TFilters
+  >({
+    data,
+    initialFilters,
+    matchesAdditionalFilters,
+  });
+
+  const { dataViewColumns, dataViewRows, pagination } = useResourceDataViewData<
+    TData,
+    TCustomRowData
+  >({
+    columns,
+    filteredData,
+    getDataViewRows,
+    showNamespaceOverride,
+    columnManagementID,
+    customRowData,
+  });
+
+  const bodyLoading = React.useMemo(
+    () => <SkeletonTableBody rowsCount={5} columnsCount={dataViewColumns.length} />,
+    [dataViewColumns.length],
+  );
+
+  const bodyEmpty = React.useMemo(
+    () => (
+      <Tbody>
+        <Tr>
+          <Td colSpan={dataViewColumns.length}>
+            <Bullseye>{t('public~No Pods found')}</Bullseye>
```

Comment:
Need to remove `Pods` from the empty state text.  Ideally this will reflect the resource name.

Reply:
```suggestion
            <Bullseye>
              {label ? t('public~No {{label}} found', { label }) : t('public~None found')}
            </Bullseye>
```
Plus the related `i18n` diff is probably better.